### DISCUSSION TITLE
PortalEngine: Public-share is empty when you try to open a public share URL

### DISCUSSION THREAD

**Discussion by khenow1978** [pimcore/pimcore#16336] [Q&A / Support]
Hi.. 

the issue I'm currently struggling with has already been brought up here: https://github.com/pimcore/pimcore/discussions/16135 Unfortunately nobody reacted so far.

However I made some investigations and might know why this error occurs, but I can't resolve it by myself.

So when you try to publically share some items from a data pool, Pimcore is supposed to insert rows in 2 database columns:

1)
<img width="702" alt="image" src="https://github.com/pimcore/pimcore/assets/92732594/3048ee18-700b-44c0-804b-f2fdd59b2a24">
The "portal_engine_public_share" contains all shares with their hash values and configs

2)
<img width="189" alt="image" src="https://github.com/pimcore/pimcore/assets/92732594/2924674b-db7a-492b-9559-d450fece190a">

The "portal_engine_public_share_item" is supposed to hold all elements which are part of the public share link. As you can see, this table is empty for the newly created link. This seems to be a behaviour since our upgrade to Pimcore 11. There are already some public shares prior to that version and those are still working.

So I checked the vendor folder and found the function that's supposed to add items to that table:

<img width="490" alt="image" src="https://github.com/pimcore/pimcore/assets/92732594/06ee64cb-a330-4b3a-a85e-23e9f7b311bd">

However it seems that the "datapoolConfig" is empty and so the object isn't set up properly and the function is never executed. This also explains the error message within the portal engine because the twig element clearly says "if the datapoolconfig ID is null show this error message" and then the "portal_engine.public_share.is_empty" message is shown.

That's the point where I can't get any further.. what exactly is the DataPoolConfig? The portal engine contains several datapools and they are all working and well configured.

Right now our users can't use the public sharing option which is quite annoying.. it would be great if someone could look into this matter. Unfortunately I have no direct access to the repository of the portal-engine and can't bring it up there.

BR,
Karsten

**Comment by fashxp** [comment]
Hi, 
please ask you contact person at Pimcore for access to the enterprise support portal. Then we can handle that on priority. 
Thank you very much...

BR
Christian

**Reply by khenow1978** [reply]
Hi Christian,

The ticket has just been created.

BR,
Karsten

**Comment by kingjia90** [comment]
Will be fixed by https://github.com/pimcore/portal-engine/pull/468 (v3.1.5)
Thank you for reporting
